## ***Prediction of Parameters Affecting the Performance of Compact Heat Exchanger***

In [ ]:
# importing libraries

import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
from sklearn.model_selection import train_test_split,GridSearchCV
from sklearn.metrics import confusion_matrix, accuracy_score, f1_score, classification_report
from sklearn import metrics #Import scikit-learn metrics module for accuracy calculation

In [ ]:
data= pd.read_csv("Training set - Sheet1.csv")
data.head(5)

mhw   Thw   Thwi   Tcw   Tcwi    mcw   Tcwo  Thwo
0    6  55.5  59.70   9.5  14.60   2.90  23.80  40.0
1    6  64.8  60.30  26.8  15.10  10.20  41.30  47.1
2    6  63.8  60.00  15.8  14.80   6.40  32.10  42.7
3    6   0.7   0.09   1.8   0.07   2.00   4.31   1.9
4    6  -1.2   0.80   3.3   1.00  -0.04   0.60   0.7

In [ ]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   mhw     42 non-null     int64  
 1   Thw     42 non-null     float64
 2   Thwi    42 non-null     float64
 3   Tcw     42 non-null     float64
 4   Tcwi    42 non-null     float64
 5   mcw     42 non-null     float64
 6   Tcwo    42 non-null     float64
 7   Thwo    42 non-null     float64
dtypes: float64(7), int64(1)
memory usage: 2.8 KB


In [ ]:
data.describe

<bound method NDFrame.describe of     mhw   Thw   Thwi   Tcw   Tcwi    mcw   Tcwo  Thwo
0     6  55.5  59.70   9.5  14.60   2.90  23.80  40.0
1     6  64.8  60.30  26.8  15.10  10.20  41.30  47.1
2     6  63.8  60.00  15.8  14.80   6.40  32.10  42.7
3     6   0.7   0.09   1.8   0.07   2.00   4.31   1.9
4     6  -1.2   0.80   3.3   1.00  -0.04   0.60   0.7
5     6   2.0   0.08   1.1   2.20  -1.10   0.50  -0.6
6     7  59.2  59.90  15.8  14.60   2.90  27.10  41.2
7     7  64.9  60.30  26.6  15.50  10.20  42.20  48.7
8     7  63.6  59.90  15.8  14.90   6.80  32.70  43.8
9     7   0.4   0.08   1.7   0.08   2.30   4.60   2.1
10    7  -1.6   0.30   2.7   1.40  -0.10   0.70   0.9
11    7   0.4   0.08   1.7   0.08   2.30  -0.50  -0.3
12    8  57.0  59.80  14.9  14.60   2.90  28.10  42.7
13    8  64.6  60.40  26.1  15.10  10.10  43.30  49.5
14    8  63.5  60.00  15.5  14.90   6.20  34.80  45.6
15    8   0.8   0.10   1.0   0.10   2.10   4.60   2.1
16    8  -3.9   0.30   2.7   0.10   0.10   0.40 

In [ ]:
X_train=data.iloc[:,0:6]

In [ ]:
X_train

mhw   Thw   Thwi   Tcw   Tcwi    mcw
0     6  55.5  59.70   9.5  14.60   2.90
1     6  64.8  60.30  26.8  15.10  10.20
2     6  63.8  60.00  15.8  14.80   6.40
3     6   0.7   0.09   1.8   0.07   2.00
4     6  -1.2   0.80   3.3   1.00  -0.04
5     6   2.0   0.08   1.1   2.20  -1.10
6     7  59.2  59.90  15.8  14.60   2.90
7     7  64.9  60.30  26.6  15.50  10.20
8     7  63.6  59.90  15.8  14.90   6.80
9     7   0.4   0.08   1.7   0.08   2.30
10    7  -1.6   0.30   2.7   1.40  -0.10
11    7   0.4   0.08   1.7   0.08   2.30
12    8  57.0  59.80  14.9  14.60   2.90
13    8  64.6  60.40  26.1  15.10  10.10
14    8  63.5  60.00  15.5  14.90   6.20
15    8   0.8   0.10   1.0   0.10   2.10
16    8  -3.9   0.30   2.7   0.10   0.10
17    8   1.4  -0.90   3.3  -1.60  -0.90
18    9  59.5  59.70  14.9  14.60   3.00
19    9  63.9  60.30  28.6  16.00  10.10
20    9  63.3  60.00  15.7  14.90   6.30
21    9   0.5   0.09   1.9   0.10   1.80
22    9  -3.9  -0.10   5.5   1.80   0.10
23    9   0.5   0.09   1.9   0.10   1.80
24   10  58.6  59.80  14.9  14.50   3.00
25   10  65.2  60.40  28.0  15.10  10.00
26   10  63.5  60.00  15.4  14.80   6.20
27   10   0.7   0.10   1.2   0.07   2.07
28   10  -2.1  -0.10   2.9   1.50   0.20
29   10   0.7   0.10   1.2   0.07   2.00
30   11  62.2  59.80  15.1  14.80   5.90
31   11  63.7  60.20  15.7  15.20   6.80
32   11  63.2  60.00  15.4  15.00   6.10
33   11   0.3   0.10   0.1   0.10   0.20
34   11  -1.1  -0.02   0.5  -0.80   0.80
35   11   0.2  -0.90  -1.3  -1.10  -1.20
36   12  60.2  59.60  15.1  14.80   2.90
37   12  63.7  60.10  29.4  15.30   8.10
38   12  63.2  59.90  16.5  14.90   5.60
39   12   0.4   0.09   3.1   0.10   1.50
40   12  -3.6   0.40   3.4   0.10  -0.30
41   12   1.3  -1.00   1.1  -0.90  -0.90

In [ ]:
X_train.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 6 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   mhw     42 non-null     int64  
 1   Thw     42 non-null     float64
 2   Thwi    42 non-null     float64
 3   Tcw     42 non-null     float64
 4   Tcwi    42 non-null     float64
 5   mcw     42 non-null     float64
dtypes: float64(5), int64(1)
memory usage: 2.1 KB


In [ ]:
temp1_train=data['Tcwo']

In [ ]:
temp2_train=data['Thwo']

In [ ]:
data1= pd.read_csv("Test set - Sheet1.csv")
data1.head(5)

mhw   Thw   Thwi   Tcw   Tcwi    mcw  Tcwo  Thwo
0    6  57.2  59.90  15.2  14.60   2.90  23.8  40.0
1    6  64.8  60.30  25.9  15.10  10.10  41.3  47.1
2    6  63.8  60.00  15.7  14.80   6.30  32.3  42.8
3    6   0.7   0.08   1.6   0.07   2.00   4.2   1.9
4    6  -2.7   0.90   2.9   1.10   0.05   0.5   0.6

In [ ]:
data1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 42 entries, 0 to 41
Data columns (total 8 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   mhw     42 non-null     int64  
 1   Thw     42 non-null     float64
 2   Thwi    42 non-null     float64
 3   Tcw     42 non-null     float64
 4   Tcwi    42 non-null     float64
 5   mcw     42 non-null     float64
 6   Tcwo    42 non-null     float64
 7   Thwo    42 non-null     float64
dtypes: float64(7), int64(1)
memory usage: 2.8 KB


In [ ]:
X_test=data1.iloc[:,0:6]

In [ ]:
temp1_test=data1['Tcwo']

In [ ]:
temp2_test=data1['Thwo']

In [ ]:
from sklearn.preprocessing import StandardScaler
scaler= StandardScaler()

X_train= scaler.fit_transform(X_train)
X_test= scaler.fit_transform(X_test)

## Multiple Linear Regression and Artificial Neural Networks Algorithms are used For the prediction of Thwo and Tcwo Values

## Multiple Linear Regression

In [ ]:
import numpy as np
from sklearn.linear_model import LinearRegression

### For prediction of Tcwo

In [ ]:
reg= LinearRegression().fit(X_train, temp1_train)

In [ ]:
output1=reg.predict(X_test)


In [ ]:
reg.score(X_train, temp1_train)

0.9936138213032514

In [ ]:
print('Accuracy of Multiple Linear Regression is:',reg.score(X_test, temp1_test))

Accuracy of Multiple Linear Regression is: 0.9217268653624169


### For prediction of Thwo

In [ ]:
reg1= LinearRegression().fit(X_train, temp2_train)

In [ ]:
reg1.score(X_train, temp2_train)

0.9976699668115995

In [ ]:
output2=reg1.predict(X_test)

In [ ]:
print('Accuracy of Multiple Linear Regression is:',reg.score(X_test, temp2_test))

Accuracy of Multiple Linear Regression is: 0.8294099503986314


## Artificial Neural Network

In [ ]:
# import required libraries
from sklearn.neural_network import MLPRegressor

### For Prediction of Tcwo 

In [ ]:
clf = MLPRegressor(hidden_layer_sizes=(100,),random_state=1, max_iter=300)

In [ ]:
clf.fit(X_train, temp1_train)

/usr/local/lib/python3.7/dist-packages/sklearn/neural_network/_multilayer_perceptron.py:696: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (300) reached and the optimization hasn't converged yet.
  ConvergenceWarning,


MLPRegressor(max_iter=300, random_state=1)

In [ ]:
clf.predict(X_test)

In [ ]:
mlp1=clf.predict(X_test)

In [ ]:
print('Accuracy of MLP ANN Regressor is:',clf.score(X_test, temp1_test))

Accuracy of MLP ANN Regressor is: 0.9839850678338016


### For prediction of Thwo

In [ ]:
clf1 = MLPRegressor(hidden_layer_sizes=(100,),random_state=1, max_iter=300)

In [ ]:
clf1.fit(X_train, temp2_train)

/usr/local/lib/python3.7/dist-packages/sklearn/neural_network/_multilayer_perceptron.py:696: ConvergenceWarning: Stochastic Optimizer: Maximum iterations (300) reached and the optimization hasn't converged yet.
  ConvergenceWarning,


MLPRegressor(max_iter=300, random_state=1)

In [ ]:
clf1.predict(X_test)

array([ 2.92441137e+01,  5.20321396e+01,  3.76998764e+01,  1.55985974e+00,
        7.62191020e-01,  1.92133404e-01,  3.07457605e+01,  4.09121929e+01,
        3.94381797e+01,  1.33578713e+00,  3.91423266e-01, -8.65345924e-02,
        3.14358269e+01,  5.27633605e+01,  3.93085835e+01,  8.78284627e-01,
        3.65276362e-01,  4.95892350e-01,  3.32207449e+01,  5.67884254e+01,
        4.06799089e+01,  7.35014476e-01,  3.44272219e-01,  2.79373643e+00,
        3.41052528e+01,  5.76460292e+01,  4.18805188e+01,  7.75927190e-01,
        3.18678218e-01, -3.59396812e-02,  4.15331997e+01,  4.41649521e+01,
        4.26569514e+01,  9.43921343e-02,  1.26172424e-01, -6.13606462e-01,
        3.68778722e+01,  5.66196621e+01,  4.35931968e+01,  6.41276021e-01,
        6.66654219e-02, -6.65774862e-02])

In [ ]:
print('Accuracy of MLP ANN Regressor is:',clf1.score(X_test, temp2_test))

Accuracy of MLP ANN Regressor is: 0.9493341619326822


## **From the results of two differnt algorithms, it can be observed that ANN is performing better in predictions as compared to Multiple Linear Regression**


### Average Accuracy of Multiple Linear Regression: 87.5 %
### Average Accuracy of ANN: 98 %